apps/gdalalg_clip_common.cpp

```diff
@@ -0,0 +1,234 @@
+                {
+                    return {
+                        nullptr,
+                        CPLSPrintf("Geometry of feature " CPL_FRMT_GIB " of %s "
+                                   "is invalid.",
+                                   poFeat->GetFID(), poDS->GetDescription())};
+                }
+                else
+                {
+                    oGC.addGeometry(std::move(poSrcGeom));
+                }
+            }
+        }
+    }
+
+    if (!m_likeSQL.empty())
+        poDS->ReleaseResultSet(poLyr);
+
+    if (oGC.IsEmpty())
+    {
+        return {nullptr, "No clipping geometry found"};
+    }
+
+    return {std::unique_ptr<OGRGeometry>(oGC.UnaryUnion()), std::string()};
+}
+
+/************************************************************************/
+/*                           GetClipGeometry()                          */
+/************************************************************************/
+
+std::pair<std::unique_ptr<OGRGeometry>, std::string>
+GDALClipCommon::GetClipGeometry()
+{
+
+    std::unique_ptr<OGRGeometry> poClipGeom;
+
+    if (!m_bbox.empty())
+    {
+        poClipGeom = std::make_unique<OGRPolygon>(m_bbox[0], m_bbox[1],
+                                                  m_bbox[2], m_bbox[3]);
+
+        if (!m_bboxCrs.empty())
+        {
+            auto poSRS = new OGRSpatialReference();
+            poSRS->SetAxisMappingStrategy(OAMS_TRADITIONAL_GIS_ORDER);
+            CPL_IGNORE_RET_VAL(poSRS->SetFromUserInput(m_bboxCrs.c_str()));
+            poClipGeom->assignSpatialReference(poSRS);
+            poSRS->Release();
+        }
+    }
+    else if (!m_geometry.empty())
+    {
+        {
+            CPLErrorStateBackuper oErrorStateBackuper(CPLQuietErrorHandler);
+            auto [poGeom, eErr] =
+                OGRGeometryFactory::createFromWkt(m_geometry.c_str());
+            if (eErr == OGRERR_NONE)
+            {
+                poClipGeom = std::move(poGeom);
+            }
+            else
+            {
+                poClipGeom.reset(
+                    OGRGeometryFactory::createFromGeoJson(m_geometry.c_str()));
+                if (poClipGeom)
+                {
+                    auto poSRS = new OGRSpatialReference();
+                    poSRS->SetAxisMappingStrategy(OAMS_TRADITIONAL_GIS_ORDER);
+                    CPL_IGNORE_RET_VAL(poSRS->SetFromUserInput("WGS84"));
+                    poClipGeom->assignSpatialReference(poSRS);
+                    poSRS->Release();
+                }
+            }
+        }
+        if (!poClipGeom)
+        {
+            return {
+                nullptr,
+                "Clipping geometry is neither a valid WKT or GeoJSON geometry"};
+        }
+
+        if (!m_geometryCrs.empty())
+        {
+            auto poSRS = new OGRSpatialReference();
+            poSRS->SetAxisMappingStrategy(OAMS_TRADITIONAL_GIS_ORDER);
+            CPL_IGNORE_RET_VAL(poSRS->SetFromUserInput(m_geometryCrs.c_str()));
```

Comment:
comment why the return doesn't need to be checked?

Reply:
"// Validity of CRS already checked by GDALAlgorithm" added